clusterloader2/pkg/measurement/common/probes/probes.go

```diff
@@ -202,28 +215,28 @@ func (p *probesMeasurement) gather(params map[string]interface{}) (measurement.S
 }
 
 func (p *probesMeasurement) createProbesObjects() error {
-	return p.framework.ApplyTemplatedManifests(p.config.Manifests, p.templateMapping)
+	return p.framework.ApplyTemplatedManifests(path.Join(manifestsPathPrefix, p.config.Manifests), p.templateMapping)
 }
 
 func (p *probesMeasurement) waitForProbesReady() error {
-	klog.Info("Waiting for Probes to become ready...")
+	klog.Infof("Waiting for Probe %s to become ready...", p)
 	return wait.Poll(checkProbesReadyInterval, checkProbesReadyTimeout, p.checkProbesReady)
 }
 
 func (p *probesMeasurement) checkProbesReady() (bool, error) {
-	serviceMonitors, err := p.framework.GetDynamicClients().GetClient().
-		Resource(serviceMonitorGvr).Namespace(probesNamespace).List(metav1.ListOptions{})
-	if err != nil {
-		if client.IsRetryableAPIError(err) || client.IsRetryableNetError(err) {
-			err = nil // Retryable error, don't return it.
-		}
-		return false, err
-	}
 	// TODO(mm4tt): Using prometheus targets to check whether probes are up is a bit hacky.
 	//              Consider rewriting this to something more intuitive.
-	expectedTargets := p.replicasPerProbe * len(serviceMonitors.Items)
+	selector := func(t prometheus.Target) bool {
+		for _, pod := range p.config.ProbeNameLabels {
+			if t.Labels["job"] == pod && t.Labels["namespace"] == probesNamespace {
```

Comment:
This won't work - the label you're setting is "probe", right?

[We should ensure that it's the same for different probes]

Reply:
It does, I've checked it specifically. Labels returned here by Prometheus are not the same as ones defined on the actual pod that Prometheus is watching.

Reply:
Ahh interesting - thanks for checking.
Can you please add a comment that prometheus does some translation here, so that someone else won't be asking the same question in the future?